src/gtk/toga_gtk/libs/style.py

```diff
@@ -0,0 +1,49 @@
+from . import Gtk
+
+
+def apply(style_context, style):
+    # creating StyleProvider (i.e CssProvider)
+    style_provider = Gtk.CssProvider()
+    style_provider.load_from_data(style.encode())
+
+    # setting the StyleProvider to StyleContext
+    style_context.add_provider(
+        style_provider,
+        Gtk.STYLE_PROVIDER_PRIORITY_USER,
+    )
+    style_context.add_class("custom-style")
+
+
+def get_color_css(value):
+    return (
+        ".custom-style {"
+        f"color: rgba({value.r}, {value.g}, {value.b}, {value.a});"
+        "}"
+    )
+
+
+def get_bg_color_css(value):
+    return (
+        ".custom-style {"
+        f"background-color: rgba({value.r}, {value.g}, {value.b}, {value.a});"
+        "background-image: none;"
+        "}"
+    )
+
+
+def get_font_css(value):
+    style = (
+        ".custom-style { "
+        f"font-style: {value.style}; "
+        f"font-variant: {value.variant}; "
+        f"font-weight: {value.weight}; "
+    )
+
+    if value.family != "system":
+        style += f"font-family: {value.family}; "
+    if value.size != -1:
+        style += f"font-size: {value.size}px; "
+
+    style += "}"
```

Comment:
Generally, it's better to combine strings using `'\n'.join(parts)` or ''.join(parts), rather than a series of `+=` calls. Repeated addition requires a copy of the original string each time an addition is made, whereas joining a series of parts means a single copy is made.

Reply:
If you look at `get_color_css` function and `get_bg_color_css` function I am not concatenating the strings. Python creates a single string at runtime. But in case of `get_font_css` function the default value for size is -1 due to which Gtk throws size can not be negative error. Therefore I am checking if size is -1, if so not adding them to the final css. Only in this specified use case I am concatenating the strings.

Reply:
I understand your concern. Will change in next commit.